src/Client/ConnectionPoolWithFailover.cpp

```diff
@@ -184,13 +185,11 @@ std::vector<ConnectionPoolWithFailover::TryResult> ConnectionPoolWithFailover::g
                             "Cannot get connection from ConnectionPoolWithFailover cause nested pools are empty");
 
     if (!skip_unavailable_endpoints.has_value())
-        skip_unavailable_endpoints = (settings && settings->skip_unavailable_shards);
+        skip_unavailable_endpoints = settings.skip_unavailable_shards;
 
     size_t min_entries = skip_unavailable_endpoints.value() ? 0 : 1;
 
-    size_t max_tries = (settings ?
-        size_t{settings->connections_with_failover_max_tries} :
-        size_t{DBMS_CONNECTION_POOL_WITH_FAILOVER_DEFAULT_MAX_TRIES});
+    size_t max_tries = settings.connections_with_failover_max_tries;
```

Comment:
I think we can remove `DBMS_CONNECTION_POOL_WITH_FAILOVER_DEFAULT_MAX_TRIES` completely. It is only used in `Settings.h`, so we can just remove the macro and use its value in `Settings.h`. This way nobody will be tempted to use the "default" number when we have a setting for this.